### PR TITLE
Add `RawArray::nulls_bitslice` with bitvec

### PR DESCRIPTION
**Pull request by workingjubilee**
This is hopefully the penultimate PR in the project to address tcdi/pgx#627.

It lets us start using the ability to handle the bitmaps for the "flat" array repr that Postgres uses internally, directly. It does not really use such, as yet, to completely outmode our usage of `deconstruct_array` where appropriate, because I found larger, more annoying obstacles in the way of changing that (see tcdi/pgx#660). I wanted to introduce this PR first to get some additional eyes on this and because it introduces some additional functionality to our new RawArray type.

cc @mhov

### CODE REVIEW

File: pgx/src/array.rs [hunk 1]
```diff
@@ -175,6 +182,21 @@ impl RawArray {
         self.data_offset() != 0
     }
 
+    /// May return null.
+    #[inline]
+    fn nulls_mut_ptr(&mut self) -> *mut u8 {
+        unsafe { pgx_ARR_NULLBITMAP(self.ptr.as_ptr()) }
+    }
```
Comment (BradyBonnette) [inline pgx/src/array.rs:189]: Does this need any `# safety` documentation? Or is it assumed here?

File: pgx/src/array.rs [hunk 2]
```diff
@@ -199,9 +221,43 @@ impl RawArray {
...
+    The [bitvec] equivalent of [RawArray::nulls].
+    If this returns `None`, the array cannot have nulls.
+    If this returns `Some`, it points to the bitslice that marks nulls in this array.
+
+    Note that unlike the `is_null: bool` that appears elsewhere, here a 0 bit is null.
+    Unlike [RawArray::nulls], this slice is bit-exact in length, so there are no caveats for safely-used BitSlices.
+
+    [bitvec]: https://docs.rs/bitvec/latest
+    [BitPtrError::Null]: <https://docs.rs/bitvec/latest/bitvec/ptr/enum.BitPtrError.html>
+    [ARR_NULLBITMAP]: <https://git.postgresql.org/gitweb/?p=postgresql.git;a=blob;f=src/include/utils/array.h;h=4ae6c3be2f8b57afa38c19af2779f67c782e4efc;hb=278273ccbad27a8834dfdf11895da9cd91de4114#l293>
```
Comment (BradyBonnette) [inline pgx/src/array.rs:237]: I love this, linking right to the postgresql source ❤️

File: pgx/src/datum/array.rs [hunk 3]
```diff
@@ -73,48 +100,82 @@ impl<'a, T: FromDatum> Array<'a, T> {
         // Remember to remove the Array::over tests in pgx-tests/src/tests/array_tests.rs
```
Comment (BradyBonnette) [inline pgx/src/datum/array.rs:100]: Does Rust have something similar to a `TODO` that can report on things like "hey check this out later"?

File: pgx/src/datum/array.rs [hunk 4]
```diff
@@ -73,48 +100,82 @@ impl<'a, T: FromDatum> Array<'a, T> {
...
+            .unwrap_or(NullKind::Strict(nelems));
+
+        Array {
             _ptr,
-            array_type,
+            raw: Some(raw),
             nelems,
-            elem_slice: slice::from_raw_parts(elements, nelems),
-            null_slice: slice::from_raw_parts(nulls, nelems),
+            elem_slice: unsafe { slice::from_raw_parts(elements, nelems) },
```
Comment (BradyBonnette) [inline pgx/src/datum/array.rs:169]: Does this need `safety` documentation?

File: pgx/src/datum/array.rs (hunk 3 above)
Comment (workingjubilee) [reply pgx/src/datum/array.rs:100]: https://docs.rs/todo-or-die/latest/todo_or_die/ ?

Comment (BradyBonnette) [reply pgx/src/datum/array.rs:100]: Harsh crate name 😆 

I was thinking maybe a one-off cargo command or something that will parse the codebase and say "you have these todos, check them out". Though, in saying it out loud in my head, that's honestly no different than a `rg TODO`

File: pgx/src/array.rs (hunk 1 above)
Comment (workingjubilee) [reply pgx/src/array.rs:189]: It does.

**Pull request merged**